lib/slapd-config.cpp

```diff
@@ -1651,6 +1651,26 @@ bool OlcDatabase::isBdbDatabase( const LDAPEntry& e )
     return false;
 }
 
+/*
+ * Return the type (frontent, config, hdb, bdb, mdb) of this database.
+ */
+std::string OlcDatabase::getDatabaseType()
+{
+    std::string attr = getStringValue("olcDatabase");
+    size_t closingBracket = attr.find_last_of('}');
+    if (closingBracket == std::string::npos)
+    {
+        // The value does not contain index, return as-is.
+        return attr;
+    }
+    else
+    {
+        // The value looks like {1234}mydb
+        // Strip index number from value
+        return attr.substr(closingBracket + 1, std::string::npos);
```

Comment:
`npos` is the default, no need to set it explicitly here (http://www.cplusplus.com/reference/string/string/substr/).